src/tools/plan-generator.ts

```diff
@@ -50,23 +47,41 @@ export class PlanGenerator {
   async createPlans(): Promise<string[]> {
     const plans: string[] = [];
     for (const recipe of this.resolvedRecipes) {
-      const particles: string[] = [];
-      for (const particle of recipe.particles) {
-        particles.push(await this.createParticle(particle));
-      }
-
-      const planArgs = [ktUtils.listOf(particles)];
-      if (recipe.annotations.length > 0) {
-        planArgs.push(PlanGenerator.createAnnotations(recipe.annotations));
-      }
-
-      const start = `object ${recipe.name}Plan : `;
-      const plan = `${start}${ktUtils.applyFun('Plan', planArgs, {startIndent: start.length})}`;
-      plans.push(plan);
+      const plan = `${ktUtils.applyFun(`val ${recipe.name}Plan = Plan`, [
+        ktUtils.listOf(await Promise.all(recipe.particles.map(p => this.createParticle(p)))),
+        ktUtils.listOf(recipe.handles.map(h => this.handleVariableName(h))),
+        PlanGenerator.createAnnotations(recipe.annotations)
+      ])}`;
+
+      plans.push([
+        ...await Promise.all(recipe.handles.map(h => this.createHandleVariable(h))),
```

Comment:
Since this list of strings contains both handle vars and plans, I think it should be renamed --
`declarations`, maybe?

Reply:
Done.